### PR TITLE
[Backport] [2.x] Ensure Jackson default maximums introduced in 2.16.0 do not conflict with OpenSearch settings (#11811)

### PR DESCRIPTION
**Pull request by reta**
Backport of https://github.com/opensearch-project/OpenSearch/pull/11811 to `2.x`

### CODE REVIEW

File: libs/x-content/src/main/java/org/opensearch/common/xcontent/XContentContraints.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    final String DEFAULT_MAX_STRING_LEN_PROPERTY = "opensearch.xcontent.string.length.max";
+    final String DEFAULT_MAX_NAME_LEN_PROPERTY = "opensearch.xcontent.name.length.max";
+    final String DEFAULT_MAX_DEPTH_PROPERTY = "opensearch.xcontent.depth.max";
+
+    final int DEFAULT_MAX_STRING_LEN = Integer.parseInt(
+        System.getProperty(DEFAULT_MAX_STRING_LEN_PROPERTY, Integer.toString(Integer.MAX_VALUE) /* no limit */ )
+    );
+
+    final int DEFAULT_MAX_NAME_LEN = Integer.parseInt(
+        System.getProperty(DEFAULT_MAX_NAME_LEN_PROPERTY, Integer.toString(Integer.MAX_VALUE) /* no limit */ )
```
Comment (reta) [inline libs/x-content/src/main/java/org/opensearch/common/xcontent/XContentContraints.java:31]: @andrross @cwperks  backporting to `2.x` with "no limit" - to preserve backward compatibility

File: server/src/main/java/org/opensearch/index/mapper/MapperService.java [hunk 2]
```diff
@@ -150,13 +151,45 @@ public enum MergeReason {
...
+                );
+            }
+        },
         Property.Dynamic,
         Property.IndexScope
     );
     public static final Setting<Long> INDEX_MAPPING_FIELD_NAME_LENGTH_LIMIT_SETTING = Setting.longSetting(
         "index.mapping.field_name_length.limit",
-        Long.MAX_VALUE,
+        Integer.MAX_VALUE,
```
Comment (reta) [inline server/src/main/java/org/opensearch/index/mapper/MapperService.java:175]: 2G per field name looks very unreasonable to be fair

**Comment by reta** [comment]
> ❌ Gradle check result for [8045c60](https://github.com/opensearch-project/OpenSearch/commit/8045c60a874a3023bf8d61fa732416c1810980eb): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/32100/)

https://github.com/opensearch-project/OpenSearch/issues/8947

**Comment by reta** [comment]
> ❌ Gradle check result for [90c74ae](https://github.com/opensearch-project/OpenSearch/commit/90c74aeef409c7021cf66a6583c1471f228ea6d5): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/32104/)

https://github.com/opensearch-project/OpenSearch/issues/11091
https://github.com/opensearch-project/OpenSearch/issues/11256
https://github.com/opensearch-project/OpenSearch/issues/9773
https://github.com/opensearch-project/OpenSearch/issues/8947
https://github.com/opensearch-project/OpenSearch/issues/8945

**Comment by reta** [comment]
> ❌ Gradle check result for [90c74ae](https://github.com/opensearch-project/OpenSearch/commit/90c74aeef409c7021cf66a6583c1471f228ea6d5): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/32106/)

https://github.com/opensearch-project/OpenSearch/issues/11256
https://github.com/opensearch-project/OpenSearch/issues/9773
https://github.com/opensearch-project/OpenSearch/issues/8947
https://github.com/opensearch-project/OpenSearch/issues/8945
https://github.com/opensearch-project/OpenSearch/issues/10983
https://github.com/opensearch-project/OpenSearch/issues/11875

**Review by kotwanikunal — APPROVED** [review]

**Pull request merged**